test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -14002,6 +13989,56 @@ def add_mom_dyn_res(self, custom_resource, script_body=None,
         self.add_config(a)
         return res_file
 
+    def enable_cgroup_cset(self):
+        """
+        Configure and enable cgroups hook
+        """
+        # check if cgroups subsystems including cpusets are mounted
+        with open(os.path.join(os.sep, 'proc', 'mounts'), 'r') as fd:
+            mounts = fd.read()
+        if 'cgroup' in mounts and mounts.count('cpuset') >= 2:
+            pbs_conf_val = self.du.parse_pbs_config(self.hostname)
+            f1 = os.path.join(pbs_conf_val['PBS_EXEC'], 'lib',
+                              'python', 'altair', 'pbs_hooks',
+                              'pbs_cgroups.CF')
+            vpn1 = r'"vnode_per_numa_node"\s*:\s*false,'
+            vpn2 = '"vnode_per_numa_node"   : true,'
+            uht1 = r'"use_hyperthreads"\s*:\s*false,'
+            uht2 = '"use_hyperthreads"      : true,'
+            mem1 = r'"memory"\s*:\s*{'
+            mem2 = r'"memsw"\s*:\s*{'
+            ena1 = r'"enabled"\s*:\s*true,'
+            ena2 = '"enabled"            : false,'
+            # set vnode_per_numa_node = true, use_hyperthreads = true
+            with open(f1, "r") as cfg:
+                lines = cfg.readlines()
+            for i, line in enumerate(lines):
+                if re.search(vpn1, line):
+                    lines[i] = re.sub(vpn1, vpn2, line)
+                if re.search(uht1, line):
+                    lines[i] = (re.sub(uht1, uht2, line))
+                # memory subsystem not mounted, do not enable in config
+                if (re.search(mem1, line) or re.search(mem2, line)) and \
+                        mounts.count('memory') == 0:
+                    if re.search(ena1, lines[i + 1]):
+                        lines[i + 1] = (re.sub(ena1, ena2, lines[i + 1]))
+            _, path = tempfile.mkstemp(prefix="cfg", suffix=".json")
+            with open(path, "w") as cfg1:
+                for line in lines:
+                    cfg1.write((line))
+            # read in the cgroup hook configuration
+            a = {'content-type': 'application/x-config',
+                 'content-encoding': 'default',
+                 'input-file': path}
+            rv = self.server.manager(MGR_CMD_IMPORT, HOOK, a,
+                                     'pbs_cgroups')
```

Comment:
If you have two moms with different features, you're going to overwrite the first one with the second one.
Do tests ever run with 2 moms, one with cpusets and one with no cpusets? It will probably fail with this PR.

Reply:
On these cgroup cpuset platforms we use one mom/mom image.

Reply:
The moms will have the same cgroups hook configuration.  Reverting all the moms are done in revert_moms().

Reply:
Will a mom on a cpuset node and a mom on a non-cpuset node work with the same cgroups configuration?